File: 剑指 Offer/剑指 Offer 03. 数组中重复的数字/Go/main3.go
func findRepeatNumber(nums []int) int {
    for index,num:=range nums{
        if index==num{
            continue
        }
        if nums[num]==num{
            return num
        }
        nums[num],nums[index] = nums[index],nums[num]
    }
    return 0
}

File: 剑指 Offer/剑指 Offer 03. 数组中重复的数字/cpp/main3.cpp
/**
* 原地置换
*/
class Solution {
public:
    int findRepeatNumber(vector<int>& nums) {
    	int n = nums.size();
        for (int i = 0; i < n; ++i) {
            while (nums[i] != i) {
                if (nums[i] == nums[nums[i]]) {
                    return nums[i];
                }
                int temp = nums[i];
                nums[i] = nums[temp];
                nums[temp] = temp;
            }
        }
        return -1;
    }
}; 